### PR TITLE
Stac v1 update

### LINKED ISSUE #466: Update STAC endpoint to be compatible with v1.0.0-r.c.1
Explorer currently uses STAC v1.0.0-beta.2. We need to update the list of conformance classes to v1.0.0-r.c.1 and properly enable item search.

Proposed list of conformance classes:
```
http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core,
http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson,
http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30,
https://api.stacspec.org/v1.0.0-rc.1/collections,
https://api.stacspec.org/v1.0.0-rc.1/core,
https://api.stacspec.org/v1.0.0-rc.1/item-search,
https://api.stacspec.org/v1.0.0-rc.1/item-search#fields,
https://api.stacspec.org/v1.0.0-rc.1/item-search#filter:basic-cql,
https://api.stacspec.org/v1.0.0-rc.1/item-search#filter:cql-json,
https://api.stacspec.org/v1.0.0-rc.1/item-search#filter:cql-text,
https://api.stacspec.org/v1.0.0-rc.1/item-search#filter:filter,
https://api.stacspec.org/v1.0.0-rc.1/item-search#filter:item-search-filter,
https://api.stacspec.org/v1.0.0-rc.1/item-search#query,
https://api.stacspec.org/v1.0.0-rc.1/item-search#sort,
https://api.stacspec.org/v1.0.0-rc.1/ogcapi-features
```

### PR DESCRIPTION
**Pull request by Ariana-B**
Building off of PR #367 to complete stac v1.0.0-r.c.1 upgrade. 

- arrivals has been moved to a catalog instead of a collection
- fixed broken code and tests (for stac only)
- reorganised `_stac.py` and `test_stac.py` for easier navigation
- updated schemas
- updated setup requirements: bump versions of datacube and eodatasets3, removed digitalearthau as it is not available through pypi

----
:books: Documentation preview :books:: https://datacube-explorer--474.org.readthedocs.build/en/474/

### CODE REVIEW

**Comment by omad** [comment]
Hi Ariana,

I haven't looked through all the code yet, but can you please have a go at fixing the errors showing up in GitHub Actions. The Slack web-hooks are fine to remove, the linting errors look pretty trivial. And it seems as though `digitalearthau` is still required for running the tests, and we do want to run tests before merging if at all possible.

Cheers

File: integration_tests/schemas/update.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ rm ${stac_tag}.tar.gz
 rm -rf "schemas.stacspec.org/${stac_version}"
 mv ${subfolder} "schemas.stacspec.org/${stac_version}"
 rm -rf "stac/${stac_version}"
-ln -s "$(pwd)/schemas.stacspec.org/${stac_version}" "$(pwd)/stac"
+ln -s "../schemas.stacspec.org/${stac_version}" "stac/"
```
Comment (omad) [inline integration_tests/schemas/update.sh:33]: This is the sort of unintuitive line it'd be worth adding a comment to. I can imagine someone tripping over it in the future.

File: README.md [hunk 2]
```diff
@@ -258,9 +258,20 @@ Install the test dependencies: `pip install -e .[test]`
...
+`docker run -p 5432:5432 -e POSTGRES_USER=<user> -e POSTGRES_PASSWORD=<password> -e POSTGRES_DB=dea_integration -d postgis/postgis`
+
+Add credentials to `.pgpass` (e.g. `localhost:5432:*:<user>:<password>`)
+
+Then:
+`export PGPASSFILE=path/to/.pgpass`
+
+`export DATACUBE_DB_URL=postgresql://<user>:<password>@localhost/dea_integration`
+
+`datacube system init`
```
Comment (omad) [inline README.md:274]: :grin:

File: integration_tests/schemas/schemas.stacspec.org/1.0.0/item-spec/json-schema/itemcollection.json [hunk 3]
```diff
@@ -30,7 +30,7 @@
                         "stac_version": {
                             "title": "STAC version",
                             "type": "string",
-                            "const": "1.0.0"
+                            "const": "0.9.0"
```
Comment (omad) [inline integration_tests/schemas/schemas.stacspec.org/1.0.0/item-spec/json-schema/itemcollection.json]: This looks like it's going the wrong way. Can you confirm it's right please.

**Review by omad — APPROVED** [review]
Looks great, thank you!

File: integration_tests/schemas/schemas.stacspec.org/1.0.0/item-spec/json-schema/itemcollection.json (hunk 3 above)
Comment (Ariana-B) [reply integration_tests/schemas/schemas.stacspec.org/1.0.0/item-spec/json-schema/itemcollection.json]: Fixed

File: integration_tests/schemas/update.sh (hunk 1 above)
Comment (Ariana-B) [reply integration_tests/schemas/update.sh:33]: Done!

**Comment by pindge** [comment]
This pr breaks stuff https://github.com/opendatacube/datacube-explorer/actions/runs/3474631398/jobs/5808010868 @Ariana-B  @omad

**Pull request merged**